tsdb/engine/tsm1/reader.go

```diff
@@ -1325,6 +1339,15 @@ func (m *mmapAccessor) init() (*indirectIndex, error) {
 		return nil, fmt.Errorf("mmapAccessor: byte slice too small for indirectIndex")
 	}
 
+	// Hint to the kernel that we will be reading the file.  It would be better to hint
+	// that we will be reading the index section, but that's not been
+	// implemented as yet.
+	if m.mmapWillNeed {
+		if err := madviseWillNeed(m.b); err != nil {
```

Comment:
munmap here? i guess this comment applies to a bunch of spots in this function lol. in addition to file handle resources. rename has the same problem. seems big enough for another ticket/bug maybe?